packages/google_sign_in/google_sign_in_platform_interface/lib/src/types.dart

```diff
@@ -26,8 +26,14 @@ enum SignInOption {
 class GoogleSignInUserData {
   /// Uses the given data to construct an instance. Any of these parameters
   /// could be null.
-  GoogleSignInUserData(
-      {this.displayName, this.email, this.id, this.photoUrl, this.idToken});
+  GoogleSignInUserData({
+    this.displayName,
+    this.email,
+    this.id,
+    this.photoUrl,
+    this.idToken,
+    this.serverAuthCode,
```

Comment:
I don't think the **User** should have information about the **server**AuthCode.

Why not limit this to the GoogleSignInTokenData so it can only be retrieved by the `getTokens()` call?

How do you want people to use this extension to the API? Can you write add a quick example/documentation on how to use the new `serverAuthCode` attribute?

Reply:
Thank you for review.

The serverAuthCode is intended to be used on iOS/Android and may also be used on Flutter for web. However, I haven't done any development on the web yet, so I haven't found any use cases on the web.

serverAuthCode usage is as follows. 

Behavior of the existing API is the same. However, in order to be able to obtain the serverAuthCode, it is necessary to define the serverClientId on the native code side. [here](https://github.com/flutter/plugins/pull/2116/files#diff-8798777ea0839080875e852fa7dca5daR81)

It is expected that the serverAuthCode can be retrieved with GoogleSignInToken Data, it is the return value of the authentication function in the GoogleSignInAccount class.
If the serverClientId was not defined, I expect the return value to be "null" when I access the attribute, because getting the serverAuthCode is optional.

In the Flutter App that I am developing, I am using the following

```
GoogleSignIn _googleSignIn = GoogleSignIn(scopes: <String>['email', etc...])
final account = await _googleSignIn.signIn();
final authentication = await account.authentication;
final serverAuthCode = authentication.serverAuthCode;
// follow serverAuthCode is used in private web service
```

Is this a reference?